docs/products/kafka/howto/terraform-governance-approvals.md

```diff
@@ -0,0 +1,240 @@
+changes using Terraform and GitHub Actions. These workflows follow three key principles:
+
+- **Topic ownership**: Apache Kafka topics are assigned to user groups using
+`owner_user_group_id`. Only members of the assigned group can approve changes to
+that topic.
+- **External identity mapping**: The `aiven_external_identity` resource links GitHub
+usernames to Aiven users, ensuring that requesters and approvers are correctly identified.
+- **Approval validation**: A GitHub Action automatically checks that pull request
+creators and approvers meet governance requirements before changes are applied.
+
+### Workflow steps:
+
+1. Define governance policies using Aiven Terraform Provider:
+
+   - Assign topic ownership (control who can approve changes) using `owner_user_group_id`.
+   - Configure user groups (`aiven_organization_user_group`) to manage approval
+     permissions.
+   - Enforce team-based approval policies to ensure compliance.
+   - Automatically validate changes against governance policies.
+
+1. Map GitHub users to Aiven identities:
+
+   - Link GitHub usernames to Aiven users using the `aiven_external_identity` resource.
+   - Ensure that only mapped users can request or approve changes.
+
+1. Submit and validate changes in GitHub
+
+   - A user submits a pull request (PR) to modify an Apache Kafka topic.
+   - The GitHub Action automatically runs governance compliance checks.
+   - Without all required approvals, the check fails.
+
+1. Approve and apply changes
+
+   - An authorized user approves the PR.
+   - The GitHub Action reruns governance validation.
+   - If all policies are met, the PR passes, and Terraform applies the changes.
+
+### Approval rules
+
+Governance enforces approval policies to maintain security and compliance:
+
+- If an approver is a member of multiple user groups, their approval applies to all teams
+  they belong to.
+- If a second approval is required, another member of the same user group must approve.
+- The pull request creator must be a member of the user group that owns the Apache
+  Kafka topic.
+
+## Set up governance approvals
+
+Set up approval workflows using Terraform and GitHub Actions.
+
+### Step 1. Define topic ownership and user groups
+
+To restrict modifications to authorized users, you must specify `owner_user_group_id`
+in your Terraform configuration:
+
+<TerraformSample filename="resources/aiven_kafka_topic/resource.tf" />
+
+Setting `owner_user_group_id` alone does not enforce approvals. The GitHub Action must
+be integrated to validate changes and enforce compliance.
+
+### Step 2. Map GitHub users to Aiven identities
+
+To verify requesters and approvers, assign them to the appropriate user group
+using the `aiven_external_identity` resource. For more information, see the
+[Aiven Terraform Provider documentation](https://registry.terraform.io/providers/aiven/aiven/latest/docs/data-sources/external_identity).
+
+**Example Terraform configuration**
+
+```hcl
```

Comment:
This whole section seems to me a good use case for inserting the example usage from the TF repo to avoid maintaining these in two places and these going out of date. I see there's no example usage for the data source, yet. I could open a PR for that today.

Reply:
Sounds good! Let me know once that example is in, and I’ll update the docs to point to it. I can also do that in a separate PR if needed; just trying to get this out ASAP.

Reply:
Totally understand that! I'll do a proper review of this after my meetings this morning and then you can just get this merged and I'll let you know when things are ready on the TF side later for updates. It usually takes a few days to get doc updates merged in the TF repo, so don't want to hold you up for that. :)